### PR TITLE
edit VirtualMachineApi.create to createOrUpdate

### PR DESCRIPTION
**Pull request by andreaturli**
- add live test for update vm

### CODE REVIEW

File: azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VirtualMachineApiLiveTest.java [hunk 1]
```diff
@@ -133,18 +135,39 @@ public void testGetInstanceView() {
...
+      List<DataDisk> oldDataDisks = oldStorageProfile.dataDisks();
+
+      ImmutableList<DataDisk> newDataDisks = ImmutableList.<DataDisk> builder().addAll(oldDataDisks).add(newDataDisk).build();
+      StorageProfile newStorageProfile = oldStorageProfile.toBuilder().dataDisks(newDataDisks).build();
+      VirtualMachineProperties newProperties = oldProperties.toBuilder().storageProfile(newStorageProfile).build();
+
+      VirtualMachine newVm = vm.toBuilder().properties(newProperties).build();
+      vm = api().createOrUpdate(vmName, newVm.location(), newVm.properties(), newVm.tags(), newVm.plan());
+
+      assertEquals(vm.properties().storageProfile().dataDisks().size(), 2);
```
Comment (aledsage) [inline azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VirtualMachineApiLiveTest.java]: Perhaps assert it's equal to `oldDataDisks.size() + 1` (or assert earlier that the oldDataDisks size is 1)?

File: azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VirtualMachineApiLiveTest.java [hunk 2]
```diff
@@ -16,11 +16,6 @@
  */
 package org.jclouds.azurecompute.arm.features;
 
-import static org.assertj.core.api.Assertions.assertThat;
```
Comment (aledsage) [inline azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VirtualMachineApiLiveTest.java]: Would be nice if the PR didn't move the imports around, but no strong feelings from me.

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/VirtualMachineApi.java [hunk 3]
```diff
@@ -88,11 +88,11 @@
    @Path("/{vmname}")
    @QueryParams(keys = "validating", values = "false")
    @Produces(MediaType.APPLICATION_JSON)
-   VirtualMachine create(@PathParam("vmname") String vmname,
-                         @PayloadParam("location") String location,
-                         @PayloadParam("properties") VirtualMachineProperties properties,
-                         @PayloadParam("tags") Map<String, String> tags,
-                         @Nullable @PayloadParam("plan") Plan plan);
+   VirtualMachine createOrUpdate(@PathParam("vmname") String vmname,
```
Comment (aledsage) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/VirtualMachineApi.java]: Could change javadoc of this method as well (not that the javadoc on most of these methods is particularly helpful, e.g. `start()`'s javadoc says "The start Virtual Machine operation"!)?

Comment (andreaturli) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/VirtualMachineApi.java]: agreed, I'll remove the trivial javadocs

**Comment by andreaturli** [comment]
thanks @nacx I will merge it to master and backport to 2.0.x

**Comment by andreaturli** [comment]
merged at [master](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/6d68b285) and [2.0.x](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/b151914f)

**Pull request closed without merging**